### PR TITLE
Ruby 3.1.2

### PR DESCRIPTION
**Pull request by aapomm**
### Summary

Upgrade Dradis ruby to 3.1.2

Updated gems:
- unicorn (`unicorn-5.4.1 requires ruby version < 3.0, which is incompatible with the current version, ruby 3.1.2p20`)
- bootsnap (`realpath_cache.rb:17:in `dirname': no implicit conversion of nil into String (TypeError)`)
- net-* gems and matrix gem (https://rubyreferences.github.io/rubychanges/3.0.html#libraries-promoted-to-default-gems)
- acts_as_tree (`NoMethodError: undefined method `arity'`)
- thor (`Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated`)
- timecop (`no implicit conversion of Hash into Integer` when running the specs)
- factory_bot (`wrong number of arguments (given 2, expected 1)` when running the specs)
- bundler-audit (`Tried to load unspecified class: Date (Psych::DisallowedClass)`)
- paper_trail (`Psych::DisallowedClass:`)

Also, `YAML.load_file` method was changed to `YAML.unsafe_load_file`

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [x] Added a CHANGELOG entry

### CODE REVIEW

File: spec/features/node_merging_spec.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@
     click_link 'Merge'
   end
 
+  after { FileUtils.rm_rf(Dir.glob(Attachment.pwd + '*')) }
```
Comment (sean-yeoh) [inline spec/features/node_merging_spec.rb:24]: Why is this change needed?

Comment (aapomm) [reply spec/features/node_merging_spec.rb:24]: In it's original location, it wouldn't trigger if the `expect` fails. Putting it in the `after` block ensures it triggers even if the test fails.

Comment (etdsoft) [reply spec/features/node_merging_spec.rb:24]: it seems we'd have this "gotcha" with failing specs elsewhere in the specs/ folder, no?

**Review by etdsoft — COMMENTED** [review]
I'm really worried about this:

> Also, YAML.load_file method was changed to YAML.unsafe_load_file

Instead of using a method that by the looks of it is unsafe, what's the recommended "safe" way to call `YAML.load` now?

File: spec/features/node_merging_spec.rb (hunk 1 above)
Comment (aapomm) [reply spec/features/node_merging_spec.rb:24]: Looks like the issue is with the rails_helper and not this. I've removed this after callback and update the cleanup in the rails_helper
